build.gradle.kts

```diff
@@ -40,17 +41,21 @@ apply(plugin = "io.github.gradle-nexus.publish-plugin")
 
 kotlin {
     explicitApi()
+
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    compilerOptions {
+        freeCompilerArgs.add("-Xexpect-actual-classes")
+    }
+
     jvm {
         compilations.all {
             // kotlin compiler compatibility options
             kotlinOptions {
-                apiVersion = "1.4"
```

Comment:
removing that means only kotlin >= 1.9 is supported, right? if it's possible to keep it explicit it's better.

But a bigger concern is that this forces all users to upgrade to kotlin 1.9, which is not that practical (yet).
If all my assumptions here are correct the best way to move forward is to merge this and create a new major version for the lib, with support for previous version for some time.

Reply:
Sadly, I can't figure out a way with lower api and language versions, but I can set them explicitly. Agreed, the best way to handle this is to create a new major version and keep support for the previous one for some time.